packages/client/tests/functional/interactive-transactions/tests.ts

```diff
@@ -762,6 +763,53 @@ testMatrix.setupTestSuite(
         `The current database provider doesn't support a feature that the query used: Mongo does not support setting transaction isolation levels.`,
       )
     })
+
+    test('should log queries that are executed inside an interactive transaction', async () => {
+      const client = newPrismaClient({
+        log: [
+          {
+            emit: 'event',
+            level: 'query',
+          },
+        ],
+      })
+
+      const queriesPromise = ((): Promise<string[]> => {
+        const entries: string[] = []
+
+        return new Promise((resolve) => {
+          client.$on('query', (data) => {
+            // @ts-expect-error
+            const query = data.query as string
+
+            entries.push(query)
+
+            if (query.includes('COMMIT')) {
+              resolve(entries)
+            }
+          })
+        })
+      })()
+
+      await prisma.$transaction(async (tx) => {
+        const user = await tx.user.create({
+          data: {
+            email: faker.internet.email(),
+          },
+        })
+
+        await tx.user.findFirst({
+          where: {
+            id: user.id,
+          },
+        })
+      })
+
+      const queries = await queriesPromise
```

Comment:
This currently times out because no query events are emitted thus causing the promise to never resolve.